onnxruntime/core/providers/webnn/webnn_execution_provider.cc

```diff
@@ -10,13 +10,101 @@
 #include "core/graph/graph_viewer.h"
 #include "core/session/onnxruntime_cxx_api.h"
 #include "core/common/safeint.h"
+#include "core/common/inlined_containers.h"
+#include "core/common/hash_combine.h"
 
 #include "builders/model.h"
 #include "builders/helper.h"
 #include "builders/model_builder.h"
 
 namespace onnxruntime {
 
+struct WebNNContextOptions {
+  std::optional<std::string> device_type;
+  std::optional<int> num_threads;
+  std::optional<std::string> power_preference;
```

Comment:
1. Is there a consideration why using string instead of enum types?

2. What about default value? For example, according to [spec](https://www.w3.org/TR/webnn/#dom-ml-createcontext), if `deviceType` is not specified, it is considered 'cpu'. However, an instance of `WebNNContextOptions` with `deviceType==="cpu"` is considered a different option of not setting `deviceType`, although they should be the same.